Frameworks/Foundation/NSOperation.mm

```diff
@@ -40,13 +40,13 @@
 #import <mutex>
 
 @implementation NSOperation {
-    StrongId<NSMutableArray> _dependencies;
+    StrongId<NSMutableSet> _dependencies;
+    NSUInteger _outstandingDependenciesCount;
     StrongId<void(^)()> _completionBlock;
     std::condition_variable_any _finishCondition;
     std::recursive_mutex _finishLock;
     std::recursive_mutex _dependenciesLock;
     std::recursive_mutex _completionBlockLock;
```

Comment:
and also what data each is protecting as it looks like _finishLock protects some non obvious members like _cancel 

---
In reply to: [72513398](https://github.com/Microsoft/WinObjC/pull/691#discussion_r72513398) [](ancestors = 72513398)